src/logic/wallets/ethTransactions.ts

```diff
@@ -38,18 +40,21 @@ export const calculateGasPrice = async (): Promise<string> => {
   return Promise.reject(err)
 }
 
-export const calculateGasOf = async (txConfig: {
-  to: string
-  from: string
-  data: string
-  gasPrice?: number
-  gas?: number
-}): Promise<number> => {
-  const web3 = getWeb3()
+export const calculateGasOf = async (txConfig: EthAdapterTransaction): Promise<number> => {
   try {
-    const gas = await web3.eth.estimateGas(txConfig)
+    const web3 = getWeb3()
+    const signerAddress = await getAccountFrom(web3)
+
+    if (!signerAddress) {
+      throw new CodedException(Errors._103, 'signerAddress returned null')
+    }
+
+    const ethAdapter = new Web3Adapter({
+      web3,
+      signerAddress,
+    })
```

Comment:
This boilerplate code will be probably repeated for each SDK call.
I feel as if this should be abstracted await in some internal utility.

Reply:
> I [suggested](https://github.com/gnosis/safe-core-sdk/issues/67) updating `Web3Adapter` to accept `null` for `signerAddress` because not every method requires it, and [we return null](https://github.com/gnosis/safe-react/blob/00c20897cf00abc13fc52a08a4cc1a89a7333740/src/logic/wallets/getWeb3.ts#L49) from `getAccountFrom`. Plus, I feel like we should just have one instance of `Web3Adapter` project-wide. What do you think? @katspaugh @germartinez

I'd suggest using a separate sdk instance for read-only stuff such as gas estimations, similarly as we do now with the web3. You can pass web3readonly instance + zero address and it should work. Accepting `null` as a `signerAddress` increases complexity for the core SDK by a lot, because it makes write methods not work (approving hash, etc)

Reply:
Makes total sense. I will remove the issue I created.

Reply:
Regarding a single SDK instance (or two instances), what would be the benefit as opposed to instantiating them on demand?
Maintaining an instance that depends on the onboard.js connection will imply hooking up to the connection life-cycle. That's an overhead I'd like to understand the trade off of.

Reply:
After discussing it with @germartinez and @katspaugh, I've abstracted the adapter instantiation into `getSDKWeb3Adapter`.